In [14]:
import os
import pandas as pd
import numpy as np
import pathlib
import xarray as xr
import rioxarray
from typing import List

In [3]:
read_path:str=r'D:\01Proj\Station-Montior-Sys\_data\perclock'
read_file:str='WS1217_DAT.08442'
full_path:str=str(pathlib.Path(read_path, read_file))
full_path

'D:\\01Proj\\Station-Montior-Sys\\_data\\perclock\\WS1217_DAT.08442'

#### 1- 读取整点风要素并生成对应集合

In [4]:
with open(full_path,'rb') as f:
    data=pd.read_csv(f, encoding='gbk', sep='\s+', header=None,
                                                infer_datetime_format=False)

In [5]:
data

0   1       2     3     4      5     6      7    8     9   ...  \
0  20231217.0  72     4.2  81.0   5.7   69.0   2.9   55.0  2.7  70.0  ...   
1        10.4  37    12.6  29.0  10.8  109.0  10.6  105.0  NaN   NaN  ...   
2         6.2  86  1013.0   NaN   NaN    NaN   NaN    NaN  NaN   NaN  ...   
3        12.6  29   324.0   NaN   NaN    NaN   NaN    NaN  NaN   NaN  ...   

     39   40    41   42    43   44    45   46    47   48  
0  75.0  3.7  54.0  2.9  77.0  3.8  68.0  2.7  78.0  4.0  
1   NaN  NaN   NaN  NaN   NaN  NaN   NaN  NaN   NaN  NaN  
2   NaN  NaN   NaN  NaN   NaN  NaN   NaN  NaN   NaN  NaN  
3   NaN  NaN   NaN  NaN   NaN  NaN   NaN  NaN   NaN  NaN  

[4 rows x 49 columns]

#### 1-1 读取第一行  
对于风要素，line0 行  
index=0 时间 eg:20231217  
index=1 -> index=48 24组数  
风向 | 风速

In [8]:
data.iloc[0][:5]

0    20231217.0
1          72.0
2           4.2
3          81.0
4           5.7
Name: 0, dtype: float64

In [15]:
data.iloc[0][0]

20231217.0

In [16]:
for index,val in enumerate(data.iloc[0][1:5]):
    print(val)

72.0
4.2
81.0
5.7


In [20]:
range(24)[0]

0

In [24]:
type(data.iloc[0])

pandas.core.series.Series

In [25]:
series_wind=data.iloc[0][1:]

In [27]:
series_wind

1     72.0
2      4.2
3     81.0
4      5.7
5     69.0
6      2.9
7     55.0
8      2.7
9     70.0
10     3.6
11    33.0
12     5.1
13    40.0
14     4.3
15    45.0
16     3.2
17    37.0
18     4.3
19    58.0
20     2.9
21    40.0
22     3.8
23    81.0
24     3.3
25    47.0
26     3.3
27    89.0
28     6.1
29    70.0
30     3.2
31    80.0
32     3.6
33    99.0
34     5.0
35    70.0
36     4.7
37    66.0
38     4.2
39    75.0
40     3.7
41    54.0
42     2.9
43    77.0
44     3.8
45    68.0
46     2.7
47    78.0
48     4.0
Name: 0, dtype: float64

In [36]:
list_wind:List[dict]=[]
for index in range(24):
    # print(index)
    step=2
    temp_wd=series_wind[index*step+1]
    temp_ws=series_wind[index*step+2]
    temp_dict={'index':index,'wd':temp_wd, 'ws':temp_ws}
    list_wind.append(temp_dict)

In [37]:
list_wind

[{'index': 0, 'wd': 72.0, 'ws': 4.2},
 {'index': 1, 'wd': 81.0, 'ws': 5.7},
 {'index': 2, 'wd': 69.0, 'ws': 2.9},
 {'index': 3, 'wd': 55.0, 'ws': 2.7},
 {'index': 4, 'wd': 70.0, 'ws': 3.6},
 {'index': 5, 'wd': 33.0, 'ws': 5.1},
 {'index': 6, 'wd': 40.0, 'ws': 4.3},
 {'index': 7, 'wd': 45.0, 'ws': 3.2},
 {'index': 8, 'wd': 37.0, 'ws': 4.3},
 {'index': 9, 'wd': 58.0, 'ws': 2.9},
 {'index': 10, 'wd': 40.0, 'ws': 3.8},
 {'index': 11, 'wd': 81.0, 'ws': 3.3},
 {'index': 12, 'wd': 47.0, 'ws': 3.3},
 {'index': 13, 'wd': 89.0, 'ws': 6.1},
 {'index': 14, 'wd': 70.0, 'ws': 3.2},
 {'index': 15, 'wd': 80.0, 'ws': 3.6},
 {'index': 16, 'wd': 99.0, 'ws': 5.0},
 {'index': 17, 'wd': 70.0, 'ws': 4.7},
 {'index': 18, 'wd': 66.0, 'ws': 4.2},
 {'index': 19, 'wd': 75.0, 'ws': 3.7},
 {'index': 20, 'wd': 54.0, 'ws': 2.9},
 {'index': 21, 'wd': 77.0, 'ws': 3.8},
 {'index': 22, 'wd': 68.0, 'ws': 2.7},
 {'index': 23, 'wd': 78.0, 'ws': 4.0}]